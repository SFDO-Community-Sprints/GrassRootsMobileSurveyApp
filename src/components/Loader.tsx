import React from 'react';
import { StyleSheet, View, Modal, ActivityIndicator, Text } from 'react-native';
import { APP_THEME, APP_FONTS } from '../constants';

type LoaderProps = {
  loading: boolean;
  message?: string;
};

function Loader(props: LoaderProps) {
  const { loading, message } = props;
  return (
    <Modal transparent visible={loading}>
      <View style={styles.modalBackground}>
        <ActivityIndicator animating={loading} size="large" hidesWhenStopped color={APP_THEME.APP_BASE_COLOR} />
        <Text style={styles.loadingTitleStyle}> {message} </Text>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  modalBackground: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: APP_THEME.APP_WHITE,
    alignItems: 'center',
    opacity: 0.8,
  },
  loadingTitleStyle: {
    fontFamily: APP_FONTS.FONT_REGULAR,
    fontSize: 36,
    color: APP_THEME.APP_BASE_COLOR,
    paddingTop: 10,
  },
});

export { Loader };
