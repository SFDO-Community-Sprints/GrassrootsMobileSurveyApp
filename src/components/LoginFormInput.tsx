import React from 'react';
import { StyleSheet } from 'react-native';
import { Input } from '@rneui/themed';
import { APP_THEME, APP_FONTS } from '../constants';

type LoginFormInputProps = {
  onChangeText(value: string): void;
  value: string;
  label: string;
  placeholder: string;
  icon: string;
  isSecure?: boolean;
  errorMessage: string;
  keyboardType?: 'email-address' | 'default';
};

function LoginFormInput(props: LoginFormInputProps) {
  return (
    <Input
      onChangeText={value => {
        props.onChangeText(value);
      }}
      value={props.value}
      label={props.label}
      placeholder={props.placeholder}
      leftIcon={{ type: 'material-community', name: props.icon, color: APP_THEME.APP_LIGHT_FONT_COLOR }}
      errorStyle={styles.errorStyle}
      labelStyle={styles.font}
      inputStyle={styles.font}
      keyboardType={props.keyboardType}
      errorMessage={props.errorMessage}
      secureTextEntry={props.isSecure}
    />
  );
}

const styles = StyleSheet.create({
  font: {
    fontFamily: APP_FONTS.FONT_REGULAR,
  },
  errorStyle: {
    color: APP_THEME.APP_ERROR_COLOR,
    fontFamily: APP_FONTS.FONT_REGULAR,
  },
});

export { LoginFormInput };
