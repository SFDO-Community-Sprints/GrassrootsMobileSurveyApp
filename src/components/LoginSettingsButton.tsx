import React from 'react';
import { Icon } from '@rneui/themed';
import { StackNavigationProp } from '@react-navigation/stack';

import { APP_THEME } from '../constants';

import { StackParamList } from '../Router';
type LoginScreenNavigationProp = StackNavigationProp<StackParamList, 'SurveyEditor'>;

type LoginSettingsButtonProps = {
  navigation: LoginScreenNavigationProp;
};

export function LoginSettingsButton(props: LoginSettingsButtonProps) {
  return (
    <Icon
      accessibilityRole="button"
      accessible
      iconStyle={{ padding: 10 }}
      name="settings"
      size={22}
      color={APP_THEME.APP_BASE_COLOR}
      type="material"
      onPress={() => {
        props.navigation.navigate('LoginSettings');
      }}
    />
  );
}
