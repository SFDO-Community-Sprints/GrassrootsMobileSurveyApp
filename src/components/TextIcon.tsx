import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Icon } from '@rneui/themed';
import { APP_FONTS } from '../constants';

type TextIconPropType = {
  icon: string;
  color: string;
  children: React.ReactNode;
};

function TextIcon(props: TextIconPropType) {
  return (
    <View style={styles.container}>
      <Icon iconStyle={{ padding: 10 }} name={props.icon} size={22} color={props.color} type="material" />
      <Text style={styles.textStyle}>{props.children}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  textStyle: {
    fontSize: 16,
    lineHeight: 22,
    fontFamily: APP_FONTS.FONT_REGULAR,
  },
});

export { TextIcon };
