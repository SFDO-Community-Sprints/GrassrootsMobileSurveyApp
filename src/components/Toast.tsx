import React from 'react';
import { BaseToast, ErrorToast } from 'react-native-toast-message';
import { APP_THEME } from '../constants';

/**
 * @description Toast config using SLDS color
 */
export const toastConfig = {
  /*
    Overwrite 'success' type,
    by modifying the existing `BaseToast` component
  */
  success: props => (
    <BaseToast
      {...props}
      style={{ borderLeftColor: APP_THEME.APP_SUCCESS_COLOR }}
      text1Style={{
        fontSize: 15,
      }}
      text2Style={{
        fontSize: 12,
      }}
    />
  ),
  /*
    Overwrite 'error' type,
    by modifying the existing `ErrorToast` component
  */
  error: props => (
    <ErrorToast
      {...props}
      style={{ borderLeftColor: APP_THEME.APP_ERROR_COLOR }}
      text1Style={{
        fontSize: 15,
      }}
      text2Style={{
        fontSize: 12,
      }}
    />
  ),
};
