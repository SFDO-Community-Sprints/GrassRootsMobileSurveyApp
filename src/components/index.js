export * from './TextHeading';
export * from './TextSubheading';
export * from './ListItem';
export * from './Loader';
export * from './TextIcon';
export * from './LoginFormInput';
