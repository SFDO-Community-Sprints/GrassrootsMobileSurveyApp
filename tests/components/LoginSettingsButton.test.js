import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import { LoginSettingsButton } from '../../src/components/LoginSettingsButton';

describe('<LoginSettingsButton />', () => {
  it('Rendering', () => {
    const screen = render(<LoginSettingsButton navigation={global.mockedNavigation} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('button can be pressed', () => {
    const screen = render(<LoginSettingsButton navigation={global.mockedNavigation} />);
    const button = screen.getByA11yRole('button');
    fireEvent.press(button);

    expect(global.mockedNavigation.navigate.mock.calls.length).toBe(1);
    expect(global.mockedNavigation.navigate.mock.calls[0][0]).toBe('LoginSettings');
  });
});
