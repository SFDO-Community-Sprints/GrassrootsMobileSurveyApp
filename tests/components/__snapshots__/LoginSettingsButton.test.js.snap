// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginSettingsButton /> Rendering 1`] = `
<View
  accessibilityRole="button"
  color="#01579b"
  iconStyle={
    Object {
      "padding": 10,
    }
  }
  name="settings"
  onPress={[Function]}
  size={22}
  type="material"
/>
`;
